spek-junit-platform-engine/src/main/kotlin/org/jetbrains/spek/engine/lifecycle/LifecycleManager.kt

```diff
@@ -4,16 +4,18 @@ import org.jetbrains.spek.api.lifecycle.ActionScope
 import org.jetbrains.spek.api.lifecycle.GroupScope
 import org.jetbrains.spek.api.lifecycle.LifecycleListener
 import org.jetbrains.spek.api.lifecycle.TestScope
-import java.util.HashSet
+import java.util.ArrayList
 
 /**
  * @author Ranie Jade Ramiso
  */
 class LifecycleManager {
-    private val listeners = HashSet<LifecycleListener>()
+    private val listeners = ArrayList<LifecycleListener>()
 
     fun addListener(listener: LifecycleListener) {
-        listeners.add(listener)
+        if (!listeners.contains(listener)) {
```

Comment:
Previously it was using a `HashedSet`, so I thought of preserving the behaviour. It's invalid to add a listener more than once, will throw an exception. 👍